pkg/controller/constrainttemplate/constrainttemplate_controller.go

```diff
@@ -133,10 +134,30 @@ func (r *ReconcileConstraintTemplate) Reconcile(request reconcile.Request) (reco
 		return reconcile.Result{}, err
 	}
 
+	instance.Status.Errors = nil
 	crd, err := r.opa.CreateCRD(context.Background(), instance)
 	if err != nil {
-		return reconcile.Result{}, err
+		var createErr *v1alpha1.CreateCRDError
+		if parseErrs, ok := err.(ast.Errors); ok {
+			for i := 0; i < len(parseErrs); i++ {
+				createErr = &v1alpha1.CreateCRDError{Code: parseErrs[i].Code, Message: parseErrs[i].Message, Location: parseErrs[i].Location.String()}
+				instance.Status.Errors = append(instance.Status.Errors, createErr)
+			}
+		} else {
+			createErr = &v1alpha1.CreateCRDError{Code: "create_error", Message: err.Error()}
+			instance.Status.Errors = append(instance.Status.Errors, createErr)
+		}
+	}
+
+	if updateErr := r.Update(context.Background(), instance); updateErr != nil {
```

Comment:
@maxsmythe 
I reorganized this part so it resets the errors to nil every time and then updates here with any errors returned by `CreateCRD`. So the `Errors` field should always consist only of the currently applicable errors and no need to filter them out later. How do you feel about this approach?

Reply:
It's fine so long as this is the only controller writing to the errors field, which is currently the case.

One last change. We need to also surface errors returned from `AddTemplate()` (lines 195 and 229 of the updated code), which is the only place compile-time errors are surfaced, such as unbound variables.